module/zfs/spa.c

```diff
@@ -984,19 +975,69 @@ spa_taskqs_init(spa_t *spa, zio_type_t t, zio_taskq_type_t q)
 		value = MIN(zio_taskq_batch_pct, 100);
 		break;
 
+	case ZTI_MODE_SCALE:
+		flags |= TASKQ_THREADS_CPU_PCT;
```

Comment:
Should we add TASKQ_DYNAMIC bit here instead?

Reply:
Why would you want it on SCALE but not on BATCH?  I personally have no opinion on this, since FreeBSD does not implement it at all.

Reply:
I thought about that, but as I mentioned in this https://github.com/openzfs/zfs/pull/11966#discussion_r632183122  I think it really makes sense to apply it to all of the current cases which is what was done.

Reply:
Ah, I missed that dynamic behavior is desirable for BATCH as well.

Change looks good.